packages/zip-it-and-ship-it/src/config.ts

```diff
@@ -3,32 +3,35 @@ import { basename, extname, dirname, join } from 'path'
 
 import isPathInside from 'is-path-inside'
 import mergeOptions from 'merge-options'
+import { z } from 'zod'
 
 import { FunctionSource } from './function.js'
-import type { NodeBundlerName } from './runtimes/node/bundlers/types.js'
-import type { ModuleFormat } from './runtimes/node/utils/module_format.js'
+import { nodeBundler } from './runtimes/node/bundlers/types.js'
+import { moduleFormat } from './runtimes/node/utils/module_format.js'
 import { minimatch } from './utils/matching.js'
 
-interface FunctionConfig {
-  externalNodeModules?: string[]
-  includedFiles?: string[]
-  includedFilesBasePath?: string
-  ignoredNodeModules?: string[]
-  nodeBundler?: NodeBundlerName
-  nodeSourcemap?: boolean
-  nodeVersion?: string
-  rustTargetDirectory?: string
-  schedule?: string
-  zipGo?: boolean
-  name?: string
-  generator?: string
-  timeout?: number
+export const functionConfig = z.object({
+  externalNodeModules: z.array(z.string()).optional().catch([]),
+  generator: z.string().optional().catch(undefined),
+  includedFiles: z.array(z.string()).optional().catch([]),
```

Comment:
Not quite sure why, but using `undefined` as the fallback for an optional value makes more sense to me.

```suggestion
  includedFiles: z.array(z.string()).optional().catch(undefined),
```